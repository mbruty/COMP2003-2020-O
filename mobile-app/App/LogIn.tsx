import React from "react";
import {
  Alert,
  Button,
  Dimensions,
  Image,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { CustomTextInput } from "./Login/CustomTextInput";
import { CONSTANT_STYLES } from "./constants";

interface Props {}

export const LogIn: React.FC<Props> = (props) => {
  // @ts-ignore
  const dimensions = Dimensions.get("window");
  const imageHeight = dimensions.height / 2;
  const imageWidth = dimensions.width;

  return (
    <ScrollView style={{ overflow: "hidden" }}>
      <Image
        style={{
          width: imageWidth,
          height: imageHeight,
          marginBottom: imageHeight,
        }}
        source={require("./log-in.png")}
      />

      <View style={[styles.card, CONSTANT_STYLES.BG_BASE_COLOUR]}>
        <Text style={[styles.titleText, CONSTANT_STYLES.TXT_RED]}>
          Track and Taste
        </Text>

        <CustomTextInput label="Email" isPassword={false} />
        <CustomTextInput label="Password" isPassword={true} />

        <TouchableOpacity>
          <Text
            style={[
              styles.highlightTxt,
              CONSTANT_STYLES.TXT_RED,
              { textAlign: "right" },
            ]}
          >
            Forgot Password?
          </Text>
        </TouchableOpacity>
        <TouchableOpacity>
          <View style={[styles.btn, CONSTANT_STYLES.BG_RED]}>
            <Text style={[styles.btnTxt, CONSTANT_STYLES.TXT_BASE]}>
              LOG IN
            </Text>
          </View>
        </TouchableOpacity>

        <TouchableOpacity style={styles.txtContainer}>
          <Text style={CONSTANT_STYLES.TXT_DEFAULT}>
            Don't have an account?
          </Text>
          <Text style={CONSTANT_STYLES.TXT_RED}>
            {" "}
            Sign Up
          </Text>
        </TouchableOpacity>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  card: {
    position: "absolute",
    top: 200,
    padding: 50,
    borderRadius: 50,
    elevation: 5,
    paddingBottom: 500,
  },
  titleText: {
    fontWeight: "bold",
    fontSize: 24,
    paddingBottom: 20,
    paddingTop: 10,
    textAlign: "center",
  },
  highlightTxt: {
    fontSize: 14,
    paddingBottom: 15,
  },
  btn: {
    paddingHorizontal: 0,
    paddingVertical: 15,
    borderRadius: 50,
  },
  btnTxt: {
    textAlign: "center",
    fontWeight: "bold",
    fontSize: 16,
  },
  txtContainer: {
    flex:1,
    flexDirection: "row",
    justifyContent: "center",
    marginTop: 25
  }
});
