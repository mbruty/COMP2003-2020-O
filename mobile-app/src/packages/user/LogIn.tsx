import React, { useState } from "react";
import {
  Dimensions,
  Image,
  KeyboardAvoidingView,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import { AwesomeTextInput } from "react-native-awesome-text-input";
import { CONSTANT_STYLES } from "../../constants";
import { PasswordInput } from "../controls";
import { SignIn } from "./utils";
import validate from "./ValidateLogin";
import LogInSvg from "../../resources/LogInSvg";

const dimensions = Dimensions.get("window");
const wHeight = dimensions.height;
const wWidth = dimensions.width;

const styles = StyleSheet.create({
  card: {
    marginTop: -wHeight * 0.7,
    padding: 50,
    borderTopLeftRadius: 50,
    borderTopRightRadius: 50,
    elevation: 5,
    width: wWidth,
    height: wHeight * 0.8,
  },
  image: {
    width: wWidth,
    height: wHeight / 2,
    marginBottom: wHeight / 2,
    marginLeft: -43,
  },
  titleText: {
    fontWeight: "bold",
    fontSize: 24,
    paddingBottom: 10,
    textAlign: "center",
  },
  highlightTxt: {
    fontSize: 14,
    paddingBottom: 15,
  },
  btn: {
    paddingHorizontal: 0,
    paddingVertical: 15,
    borderRadius: 50,
  },
  btnTxt: {
    textAlign: "center",
    fontWeight: "bold",
    fontSize: 16,
  },
  txtContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center",
    marginTop: 25,
  },
});

interface Props {
  submit: () => void;
  setPage: React.Dispatch<React.SetStateAction<string>>;
}

const LogIn: React.FC<Props> = ({ submit, setPage }) => {
  const [values, setValues] = useState<SignIn>({
    email: "",
    password: "",
  });
  const [validated, setValidated] = useState<SignIn>({
    email: "",
    password: "",
  });

  return (
    <View style={{ width: wWidth, height: wHeight }}>
      <LogInSvg style={styles.image} />
      <KeyboardAvoidingView
        style={[styles.card, CONSTANT_STYLES.BG_BASE_COLOUR]}
      >
        <Text style={[styles.titleText, CONSTANT_STYLES.TXT_RED]}>
          Track and Taste
        </Text>

        {validated.logInFailed && (
          <Text
            style={[
              CONSTANT_STYLES.TXT_RED,
              {
                fontWeight: "bold",
                textAlign: "center",
                fontSize: 16,
                paddingTop: 10,
              },
            ]}
          >
            Invalid Login
          </Text>
        )}
        <AwesomeTextInput
          autoCapitalize="none"
          keyboardType="email-address"
          customStyles={{
            title: CONSTANT_STYLES.TXT_DEFAULT,
            container: { marginTop: 25 },
          }}
          onChangeText={(text) => setValues({ ...values, email: text })}
          label="Email"
        />
        <Text style={CONSTANT_STYLES.TXT_RED}>{validated.email}</Text>
        <PasswordInput
          label="Password"
          customStyles={{
            title: CONSTANT_STYLES.TXT_DEFAULT,
            container: { marginTop: 25 },
          }}
          onChangeText={(text) => setValues({ ...values, password: text })}
        />
        <Text style={CONSTANT_STYLES.TXT_RED}>{validated.password}</Text>
        <TouchableOpacity>
          <Text
            style={[
              styles.highlightTxt,
              CONSTANT_STYLES.TXT_RED,
              { textAlign: "right", marginTop: 10 },
            ]}
          >
            Forgot Password?
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => {
            validate(values, submit).then((errors) => {
              if (errors) {
                setValidated(errors);
              }
            });
          }}
        >
          <View style={[styles.btn, CONSTANT_STYLES.BG_RED]}>
            <Text style={[styles.btnTxt, CONSTANT_STYLES.TXT_BASE]}>
              LOG IN
            </Text>
          </View>
        </TouchableOpacity>

        <TouchableOpacity
          style={styles.txtContainer}
          onPress={() => setPage("sign-up")}
        >
          <Text style={[CONSTANT_STYLES.TXT_DEFAULT]}>
            Don't have an account?
          </Text>
          <Text style={CONSTANT_STYLES.TXT_RED}> Sign Up</Text>
        </TouchableOpacity>
      </KeyboardAvoidingView>
    </View>
  );
};

export default LogIn;
