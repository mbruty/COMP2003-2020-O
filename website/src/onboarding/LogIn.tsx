import { Paper } from "@material-ui/core";
import React from "react";
import LogInStepper from "./LogInSteeper";
import SignInForm from "./SignInOrUpForm";

interface Props {
  refresh: () => void;
}

const LogIn: React.FC<Props> = (props) => {
  const [activeStep, setActiveStep] = React.useState<number>(0);
  const [tabIdx, setTabIdx] = React.useState<number>(0);

  const renderPage = () => {
    switch (activeStep) {
      case 0:
        return (
          <div className="log-in">
            <SignInForm
              tabIdx={tabIdx}
              setTabIdx={setTabIdx}
              refresh={props.refresh}
            />
          </div>
        );
      default:
        return null;
    }
  };

  // If both active step and tabidx are 0, then we're loggin in and don't need to show the stepper
  const showStepper = !(tabIdx === 0 && activeStep === 0);

  return (
    <div className="content">
      <Paper style={{ padding: 25 }}>
        <div>
          <h1>Welcome to the Track and Taste Dashboard</h1>
        </div>
        {renderPage()}
        <LogInStepper
          show={showStepper}
          activeStep={activeStep}
          setActiveStep={setActiveStep}
        />
      </Paper>
    </div>
  );
};

export default LogIn;
