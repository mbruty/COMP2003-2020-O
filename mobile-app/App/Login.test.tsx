import React from "react";
import renderer from "react-test-renderer";
import { LogIn } from "./LogIn";

test("Renders Circle Button List", () => {
  const logInPage = renderer.create(<LogIn />).toJSON();
  expect(logInPage).toMatchSnapshot();
});
