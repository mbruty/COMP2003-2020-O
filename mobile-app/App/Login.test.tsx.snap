// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Circle Button List 1`] = `
<RCTScrollView
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 667,
          "marginBottom": 667,
          "width": 750,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 50,
            "elevation": 5,
            "padding": 50,
            "paddingBottom": 500,
            "position": "absolute",
            "top": 200,
          },
          Object {
            "backgroundColor": "#F1F1F1",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "paddingBottom": 20,
              "paddingTop": 10,
              "textAlign": "center",
            },
            Object {
              "color": "#FD4040",
            },
          ]
        }
      >
        Track and Taste
      </Text>
      <View
        style={
          Object {
            "marginBottom": 20,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "left": 10,
                "position": "absolute",
                "top": 12,
              }
            }
          >
            Email
          </Text>
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#c4c4c4",
                "borderRadius": 5,
                "padding": 10,
                "width": "100%",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              rejectResponderTermination={true}
              secureTextEntry={false}
              testID="input"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
            "width": 650,
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "left": 10,
                "position": "absolute",
                "top": 12,
              }
            }
          >
            Password
          </Text>
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#c4c4c4",
                "borderRadius": 5,
                "padding": 10,
                "width": "100%",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              rejectResponderTermination={true}
              secureTextEntry={true}
              testID="input"
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 5,
              "top": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 25,
                "width": 25,
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 14,
                "paddingBottom": 15,
              },
              Object {
                "color": "#FD4040",
              },
              Object {
                "textAlign": "right",
              },
            ]
          }
        >
          Forgot Password?
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 50,
                "paddingHorizontal": 0,
                "paddingVertical": 15,
              },
              Object {
                "backgroundColor": "#FD4040",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#F1F1F1",
                },
              ]
            }
          >
            LOG IN
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 25,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#707070",
            }
          }
        >
          Don't have an account?
        </Text>
        <Text
          style={
            Object {
              "color": "#FD4040",
            }
          }
        >
           
          Sign Up
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
